File: src/dataset/typescript/reverse-string_Exercism-typescript-FilipeCerejo/reverse-string.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const reverse_string_1 = require("./reverse-string");
describe('Reverse String', () => {
    it('an empty string', () => {
        const expected = '';
        expect((0, reverse_string_1.reverse)('')).toEqual(expected);
    });
    xit('a word', () => {
        const expected = 'tobor';
        expect((0, reverse_string_1.reverse)('robot')).toEqual(expected);
    });
    xit('a capitalized word', () => {
        const expected = 'nemaR';
        expect((0, reverse_string_1.reverse)('Ramen')).toEqual(expected);
    });
    xit('a sentence with punctuation', () => {
        const expected = `!yrgnuh m'I`;
        expect((0, reverse_string_1.reverse)(`I'm hungry!`)).toEqual(expected);
    });
    xit('a palindrome', () => {
        const expected = 'racecar';
        expect((0, reverse_string_1.reverse)('racecar')).toEqual(expected);
    });
});

File: src/dataset/cpp/reverse_string_Exercism_viniciusjavs/test/reverse_string_test.cpp
#include "../src/reverse_string.h"
#include "test/catch.hpp"

TEST_CASE("an_empty_string")
{
    REQUIRE("" == reverse_string::reverse_string(""));
}

TEST_CASE("a_word")
{
    REQUIRE("tobor" == reverse_string::reverse_string("robot"));
}

TEST_CASE("a_capitalized_word")
{
    REQUIRE("nemaR" == reverse_string::reverse_string("Ramen"));
}

TEST_CASE("a_sentence_with_punctuation")
{
    REQUIRE("!yrgnuh m'I" == reverse_string::reverse_string("I'm hungry!"));
}

TEST_CASE("a_palindrome")
{
   REQUIRE("racecar" == reverse_string::reverse_string("racecar"));
}
